src/TranslationResult.h

```diff
@@ -8,101 +8,178 @@
+  /// sparse-matrix constructions with indices (Alignment).
+
+  /// Returns quality score associated with the i-th pair.
+  const Quality &quality(size_t sentenceIdx) const {
+    return scores_[sentenceIdx];
+  }
+
+  /// Returns alignment information for the i-th pair.
+  const Alignment &alignment(size_t sentenceIdx) const {
+    return alignments_[sentenceIdx];
+  }
+
+  const std::string_view getOriginalText() const {
+    std::cerr << "DeprecationWarning: getOriginalText() is deprecated in "
+                 "favour of source() to "
+                 "maintain consistency."
+              << std::endl;
+    std::string_view sourceView(source_.blob);
+    return sourceView;
+  }
+
+  const std::string_view getTranslatedText() const {
+    std::cerr << "DeprecationWarning: getTranslatedText() is deprecated in "
+                 "favour of translation() to "
+                 "maintain consistency."
+              << std::endl;
+    std::string_view translatedView(translation_.blob);
+    return translatedView;
+  }
+
+  typedef std::vector<std::pair<const std::string_view, const std::string_view>>
+      SentenceMappings;
+
+  SentenceMappings getSentenceMappings() const {
+    SentenceMappings mappings;
+    for (size_t idx = 0; idx < size(); idx++) {
+      mappings.emplace_back(source_sentence(idx), translated_sentence(idx));
+    }
+    return mappings;
+  };
+
 private:
-  // Original text (in UTF-8 encoded format) that was supposed to be translated
-  std::string originalText;
-
-  // Translation (in UTF-8 encoded format) of the originalText
-  std::string translatedText;
-
-  // Quality score of the translated text at the granularity specified in
-  // TranslationRequest. It is an optional result (it will have no information
-  // if not requested in TranslationRequest)
-  QualityScore qualityScore;
-
-  // Information regarding how individual sentences of originalText map to
-  // corresponding translated sentences in joined translated text
-  // (translatedText) An example of sentence mapping:
-  //     originalText (contains 2 sentences)              = "What is your name?
-  //     My name is Abc." translatedText (contains 2 translated sentences) =
-  //     "Was ist dein Name? Mein Name ist Abc." sentenceMappings = [
-  //         {"What is your name?", "Was ist dein Name?"},  //
-  //         Pair(originalText[0],translatedText[0])
-  //         {"My name is Abc", "Mein Name ist Abc."}       //
-  //         Pair(originalText[1],translatedText[1])
-  //     ]
-  //
-  // It is an optional result (it will be empty if not requested in
-  // TranslationRequest).
-  SentenceMappings sentenceMappings;
+  AnnotatedBlob source_;
+  AnnotatedBlob translation_;
+  std::vector<Alignment>
+      alignments_; // Indices provided by alignments are consistent with the
+                   // word annotations for the i-th source and target
+                   // sentence.
+  std::vector<Quality>
```

Comment:
@jerinphilip A quick question. Why do we want to replace `QualityScore qualityScore;` with `std::vector<Quality> scores_`?

Reply:
If you mean [this](https://github.com/browsermt/bergamot-translator/blob/2c1515313e5b6ecacbf7b4870f26abc0b82cd896/src/QualityScore.h), they're different enough to not try to patch one into the other? I honestly couldn't make of what that class intended. Really weird way to construct a tagged union/varian/sum type.

Reply:
Now that I am clear on what `Quality` struct does, I have few questions.

We already have a [QualityScore](https://github.com/browsermt/bergamot-translator/blob/jp/word-alignments/src/QualityScore.h#L22) class whose purpose is to return quality estimates for the translated text.

Are you proposing to replace that class with `Quality` struct?
If yes, then could you please list down the reasons of doing so? I would just like to avoid reinventing the wheel 👍


To explain a bit more about [QualityScore](https://github.com/browsermt/bergamot-translator/blob/jp/word-alignments/src/QualityScore.h#L22) class:
1. [textViews](https://github.com/browsermt/bergamot-translator/blob/jp/word-alignments/src/QualityScore.h#L25) represent the views in translated blob. When [textGranularity](https://github.com/browsermt/bergamot-translator/blob/jp/word-alignments/src/QualityScore.h#L31) is [sentence-level/word-level](https://github.com/browsermt/bergamot-translator/blob/jp/word-alignments/src/QualityScore.h#L15-L16), then `textViews` represent views of the sentences/words in blob. 
2. [textScores](https://github.com/browsermt/bergamot-translator/blob/jp/word-alignments/src/QualityScore.h#L28) are the scores corresponding to `textViews`

Is it not enough to represent quality score?

Reply:
> Are you proposing to replace that class with Quality struct?

Yes.

> If yes, then could you please list down the reasons of doing so?

I don't know how to construct an incomplete QualityScore with private members and no constructor, little or no documentation. Added complexity of a granularity as a tag to create a variant/tagged-union which is injected through TranslationRequest (which is a dummy and incomplete again). I don't want to carry this much overhead for nothing. Seeing as QualityScore (which could not have been constructed given the state it is in anyway) is not exported to WASM, what's there to lose in discarding it? Also, it still has string_views, which you're discarding for WASM anyway per https://github.com/mozilla/bergamot-translator/issues/8. Quality doesn't.

> Is it not enough to represent quality score?

Evidently not.

Overall, your suggestion puts me in the kill Unified API layer camp, which I'm adding to plenary agenda.